cppcache/src/util/size_tracking_allocator.hpp

```diff
@@ -0,0 +1,110 @@
+/*
```

Comment:
This is so cool and probably worth utilizing internally for object size counting in the future. 👏